### PR TITLE
Add documentation for Google PubSub Consumer

### PR DESCRIPTION
**Pull request by Agares**
Signed-off-by: Ramona Luczkiewicz <rluczkiewicz@wayfair.com>

### CODE REVIEW

**Review by Licenser — COMMENTED** [review]
nice! :D

File: docs/reference/connectors/gpubsub_consumer.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+## Metadata
+The connector will set the `$pubsub_connector` metadata variable, which is a dictionary of the messages metadata.
+
+| field        | type                    | description                                                                                 |
+|--------------|-------------------------|---------------------------------------------------------------------------------------------|
+| message_id   | String                  | The ID of the message, as provided by PubSub                                                |
+| ordering_key | String                  | The ordering key of the message                                                             |
+| publish_time | u64                     | The time when the message was published (as nanoseconds since 1st January 1970 00:00:00 UTC |
+| attributes   | HashMap<String, String> | The attributes attached to the message                                                      |
```
[the remark this answers is not in the document]
Comment (Agares) [reply docs/reference/connectors/gpubsub_consumer.md:35]: sure, what would the names here be? I guess `String` stays, `u64` -> `integer`, `HashMap<String, String>` -> `object with string values`?

[the remark this answers is not in the document]
Comment (Licenser) [reply docs/reference/connectors/gpubsub_consumer.md:35]: Ja, so it's usually called a `record` as object holds a bit of a undertone of it having more then just data so a `record with string values` would be perfect :)

**Review by Licenser — APPROVED** [review]
🚀 great! :D

File: docs/reference/connectors/gpubsub_consumer.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+| field        | type                      | description                                                                                 |
+|--------------|---------------------------|---------------------------------------------------------------------------------------------|
+| message_id   | string                    | The ID of the message, as provided by PubSub                                                |
+| ordering_key | string                    | The ordering key of the message                                                             |
+| publish_time | integer                   | The time when the message was published (as nanoseconds since 1st January 1970 00:00:00 UTC |
+| attributes   | record with string values | The attributes attached to the message                                                      |
+
+## Payload structure
+The raw payload will be passed as is to the codec
```
Comment (mfelsche) [inline docs/reference/connectors/gpubsub_consumer.md:38]: Suggestion:

The raw payload will be passed as is to the configured preprocessors and codec.

File: docs/reference/connectors/gpubsub_consumer.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        "connect_timeout": 100000000, # optional - Connection timeout in nanoseconds, defaults to 1 second 
+        "ack_deadline": 100000000, # optional - ACK deadline in nanoseconds, defaults to 10 seconds. PubSub will resend the message if it's not ACKed within this time
+        "subscription_id": "projects/my_project/subscriptions/test-subscription-a", # required - ID of the subscription to use
+        "endpoint": "https://us-east1-pubsub.googleapis.com" # optional - the endpoint for the PubSub API, defaults to https://pubsub.googleapis.com
+    }
+end;
+```
+
+## Metadata
+The connector will set the `$pubsub_connector` metadata variable, which is a dictionary of the messages metadata.
```
Comment (mfelsche) [inline docs/reference/connectors/gpubsub_consumer.md:28]: dictionary ==> recird

**Comment by mfelsche** [comment]
ok, i was too late

**Pull request merged**